### ISSUE TITLE
Plugin Installation and Mermaid Usage Instructions Missing

### ISSUE THREAD

**Issue by startakovsky** [jupyterlab/jupyterlab#15139]
## Description

This is a documentation-only issue, and I've spent some time digging to get to the bottom of this but there is nothing I've been able to find on the web, phind.com, or, importantly, Jupyter Lab's documentation.

1. [Here](https://jupyterlab.readthedocs.io/en/stable/user/extensions.html#id9) is the Installation Instructions for Extensions, they seem incomplete because they don't have an example of the syntax of how to install a plugin. While it may be as simple as pip installing, it's unclear how to treat the @jupyterlab/extension syntax.

2. Additionally, this [PR](https://github.com/jupyterlab/jupyterlab/issues/14788) has been merged and from what I can tell is out however even on the [latest](https://jupyterlab.readthedocs.io/en/latest/search.html?q=mermaid) documentation (as opposed to stable) and mermaid is listed as a core plugin, but without instructions.

It would be great to know some tips and tricks and usage instructions. For example, is this applicable in markdown files and notebook files? Is there a way to make the mermaid diagram source available while viewing a markdown cell. Often with plotting libraries you can see the code that generates the rendered plot. Is there something like that for a markdown cell. Is this also available on jupyter notebook, or just in jupyter lab?

Thank you.

**Comment by krassowski** [comment]
Note: this feature was not yet released. No installation is required as it will be included by default in JupyterLab 4.1. If you attempt installing it on older version it may not work and it is not supported. You can try it out using 4.1 alpha pre-release, currently [4.1.0a1](https://github.com/jupyterlab/jupyterlab/releases/tag/v4.1.0a1).

Writing docs for it would be a great first contribution. If you would like to do so, it could just a simple paragraph in User Guide, or as much as a complete test case generating automated screenshot (this is how we ensure that pictures in docs are up-to date).

Yes, we should include Mermaid in user-facing documentation before release. Maybe we could have something like [GitHub does here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams#creating-mermaid-diagrams).

> For example, is this applicable in markdown files and notebook files?

Yes.

> Is there a way to make the mermaid diagram source available while viewing a markdown cell

Kind of. For notebooks click on the tab and choose "New view for Notebook", drag it to the side and then you can edit code in one copy, and view result in the other:

![preview](https://github.com/jupyterlab/jupyterlab/assets/5832902/b5b7110c-e505-4e2f-ac58-11633b315b56)

For files use "Show markdown preview" from context menu. You can also copy source code from context menu.

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/c7dc089f-4709-4efc-aefa-22931ca6c6fc)

> Is this also available on jupyter notebook, or just in jupyter lab?

It likely will be supported in Notebook 7.1+

**Comment by startakovsky** [comment]
Cool, yes I have used the Pipfile below and it works just as you've described. Awesome to see.

How did you generated those animations up there?  When is the earliest possible release date for this (so that I have an idea of when to complete the documentation if I do contribute)?

```bash
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
jupyterlab = "==4.1.0a1"
nbconvert = "*"

[dev-packages]

[requires]
python_version = "3.11"
python_full_version = "3.11.4"
```

**Comment by krassowski** [comment]
I am using `peek` for gif recording; we do not have an automated workflow for gifs in documentation.

> When is the earliest possible release date

Frankly, some users are still in process in migrating to JupyterLab 4.0 as the migration of extensions is nearing completion (only LSP and git extensions remain from the major ones, with LSP getting a release this week). I would expect the major focus to shift towards 4.1 by the end of next month and hope we can get a release candidate by November.

**Comment by bollwyvl** [comment]
Yep, generally we need some content that covers what parts of the relevant upstream specs we cover. Not just `~~~mermaid`, but:

- basic markdown
- LaTeX _and_ limitations
- GFM (or the relevant parts of it)
- GitHub Markdown-as-she-is-wrote
  - mermaid 

...and ends up: 

- in the docs (with a appropriate cross-links)
- in the shipped application, available from...
  - the command palette
  - the _Help_ menu
  - the context menu of an `text/markdown`

This should be openable as a sidebar, such that it works well downstream with `notebook 7`. Given the real estate, though, examples might be preferred to be side-by-side, rather than top-bottom, but given the sidebar constraint, probably needs to be pretty responsive.

Ideally, and this has come up in the `jupyter-lsp` work, it would be ideal if kernels could _also_ provide this kind of content, documenting in a discoverable way their known divergence from the "standard" syntax of a mimetype, e.g. `text/x-ipython`'s `%%time`. 

So while the kernel-based one doesn't currently exist, we can already think about a good pattern. At present, the `latex-extension`, `mermaid-extension`, and `markdownviewer-extension` would need to register their syntax. While this _could_ be done directly in the settings schema, and would be pretty cool, this would bloat the sometimes already-large files, which get loaded every time from disk and over the wire... and writing complex markdown/latex in `.json` would be.... awful.

Centralizing the implementation around something extensible by mimetype and document e.g. 

```ts
export interface ISyntaxHelpManager {
  register(options: ISyntaxHelpManager.IRegisterOptions): void;
}

export namespace ISyntaxHelpManager {
  export interface IRegisterOptions {
    mimetypes: string[];
    syntaxGetter(options: IHelpOptions): Promise<ISyntaxHelp>;
  }
  export interface IHelpOptions {
    widget: Widget;
    translator: ITranslator;
  }
  export interface ISyntaxHelp>; {
    sections: Record<string, ISyntaxHelpSection>;
  }
  export interface ISyntaxHelpSection {
    title: string;
    rank: number;
    description: string;
    examples: IHelpExample;
    references: Record<string, string>; 
  }
  export interface IHelpExample {
    description: string;
    source: string;
    output: string;
  }
}
```

**Comment by bollwyvl** [comment]
Actually, for the special case of `text/x-ipython`, without a language server, it could _perhaps_ fetch all the completions for `%`, `%%` and `!`, then fetch signatures. would be a lotta messages, but would be relatively lazy.

**Comment by krassowski** [comment]
Just to say that the https://github.com/jupyterlab/jupyterlab/labels/good%20first%20issue part here would be solved by a PR which adds a short description of Markdown features in the user documentation (https://github.com/jupyterlab/jupyterlab/tree/main/docs/source/user), and does not need to tackle more advanced stuff suggested by Nick above (of course it would be nice to mention features other than Markdown alone!).

**Comment by dev-amanydv** [comment]
Hi, I'd love to contribute to this issue as my first open-source PR. It looks like a great place to get started with documentation, and I'm comfortable working with Markdown and exploring JupyterLab to test Mermaid usage.

Can I please be assigned this issue?

**Comment by github29tani** [comment]
Hi! I'd like to work on this Mermaid documentation issue for GSoC 2026 preparation.

I plan to:
1. Create comprehensive Mermaid usage documentation
2. Add visual examples and screenshots
3. Document installation and syntax
4. Include tips for markdown and notebook usage

I can submit a PR within 3-5 days. Could you assign this to me?

Thanks!

**Comment by jasongrout** [comment]
I think we don't assign issues based on promises of PRs, but you are more than welcome to submit a PR!

**Comment by anupamdwivedi98** [comment]
Hi, I would like to work on this documentation issue.
Could you please confirm where the Mermaid usage instructions should be added?

Thanks!

**Comment by anupamdwivedi98** [comment]
Hi, I would like to work on this documentation issue.
Could you please confirm where the Mermaid usage instructions should be added?

Thanks!

**Comment by krassowski** [comment]
We have four PRs open for this, we do not need more.
